templates/kops-create.sh.j2

```diff
@@ -61,22 +61,17 @@ echo
 
 
 echo "#------------------------------------------------------------"
-echo "# 1/4 Create Kops configuration on S3"
+echo "# 1/2 Create Kops configuration on S3"
 echo "#------------------------------------------------------------"
 
 echo "kops create --state ${KOPS_STATE_STORE} -f cluster.yml"
 if [ "${RUN}" -eq "1" ]; then
 	kops create --state ${KOPS_STATE_STORE} -f "${SCRIPTPATH}/cluster.yml"
 fi
 
-echo "kops create --state ${KOPS_STATE_STORE} -f ig-master.yml"
+echo "kops create --state ${KOPS_STATE_STORE} -f instance-groups.yml"
 if [ "${RUN}" -eq "1" ]; then
-	kops create --state ${KOPS_STATE_STORE} -f "${SCRIPTPATH}/ig-master.yml"
-fi
-
-echo "kops create --state ${KOPS_STATE_STORE} -f ig-worker.yml"
-if [ "${RUN}" -eq "1" ]; then
-	kops create --state ${KOPS_STATE_STORE} -f "${SCRIPTPATH}/ig-worker.yml"
+	kops create --state ${KOPS_STATE_STORE} -f "${SCRIPTPATH}/instance-groups.yml"
 fi
 
 echo "kops create secret --state ${KOPS_STATE_STORE} --name ${NAME} sshpublickey admin -i ./ssh-key.pub"
```

Comment:
I think you have to create secret before creating instance-groups (because of bastion host inside)

Reply:
This works fine. All those commands do are updating the configuration on S3. For that it should not matter in what order. The key is to have the following command at the very end, which in fact reads the configuration from S3 and then applies it:

```
kops update cluster --state ${KOPS_STATE_STORE} --name ${NAME} --yes
```

Reply:
Correct.